src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +1001,41 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn lockfile_path(
+    lockfile_path: Option<&Path>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<PathBuf>> {
+    let Some(lockfile_path) = lockfile_path else {
+        return Ok(None);
+    };
+
+    gctx.cli_unstable()
+        .fail_if_stable_opt("--lockfile-path", 5707)?;
+
+    let path = gctx.cwd().join(lockfile_path);
+
+    if !path.ends_with(LOCKFILE_NAME) && !crate::util::toml::is_embedded(&path) {
```

Comment:
I think it _should_ work. I initially added a basic test for it https://github.com/Ifropc/cargo/commit/164aae02d27339c6e5f33e1a4c619e0c4ed79bef but removed it since (not sure why actually). 
Do you think we would need a better test coverage for `-Zscript`? If yes, I'd prefer to cover it with tests in a separate PR. If no, I can just bring the test back.
(FYI: there's a missing `.no_manifest()` on 
```rust
  let p = project()
        .no_manifest()
```  
But it still works with adding `.no_manifest()`)

Reply:
I mean. `path` not ending with `Cargo.lock` is already emitting error. Why we need to check if it is not embedded (not ends with `.rs`)?

For `-Zscript` support, I think have a test for `cargo run` with both `--lockfile-path` and `--manifest-path` provided should cover it.

Reply:
Yeah that makes sense, I think I got accidentally mixed logic for lockfile directory in there, not sure why this check is still there, good catch! I will also cover it with a test case ^